app/_data/docs_nav_deck_1.12.x.yml

```diff
@@ -0,0 +1,86 @@
+product: deck
+version: 1.12.x
+generate: true
+items:
+- title: Introduction
+  icon: /assets/images/icons/documentation/icn-flag.svg
+  url: /deck/
+  src: index
```

Comment:
Would we have to set `src: index` for any page that has the name index.md? Or does this just replace `absolute_url: true` ?

Reply:
We'd need to set `src: index` for any index page based on the current implementation. The logic is as follows:

* If there's a `src` key, look for the `src/<product>/<src>` file
* If that doesn't exist, look for the `src/<product>/<src>/index.md` file
* If that doesn't exist, look in the `src/<product>/<url>` file
* If that doesn't exist, look in the `src/<product>/<url>/index.md` file

With `/deck/` as the URL, it would look for `src/deck/deck.md` and `src/deck/deck/index.md`, neither of which exist. I _could_ add an additional check for `src/<product>/index.md` as a fallback, but I think that may lead to unexpected behaviour. Setting `src` is much more explicit.

The generator uses `url` to generate the output path too, so the following example:

```
product: gateway
version: 1.2.x
items:
  - title: Example
     src: konnect/choose-cloud
     url: /next-steps/cloud
```

Would read `src/gateway/konnect/choose-cloud.md` and write it out at `/gateway/1.2.x/next-steps/cloud`

Any entries that have `absolute_url` set assume that the page will be generated another way (perhaps by the same code, but by another data file) and is ignored